File: src/services/horarioService.js
import { get, post, put, remove } from './api'

export const getHorariosByEmpresa = (idEmpresa) => get(`/horarios/empresa/${idEmpresa}`)
export const createHorario = (datos) => post('/horarios', datos)
export const updateHorario = (id, datos) => put(`/horarios/${id}`, datos)
export const deleteHorario = (id) => remove(`/horarios/${id}`)

File: lib/services/horario_service.dart
import 'dart:convert';
import 'package:reservapp_mobile/models/horario.dart';
import 'package:reservapp_mobile/utils/api_client.dart';

class HorarioService {
  Future<List<Horario>> getHorariosByEmpresa(int idEmpresa) async {
    try {
      final response = await ApiClient.get('/horarios/empresa/$idEmpresa');
      if (response.statusCode == 200) {
        return (jsonDecode(response.body) as List)
            .map((j) => Horario.fromJson(j))
            .toList();
      }
      return [];
    } catch (_) {
      return [];
    }
  }
}
